File: novel-backend/novel-service/src/main/java/com/novel/controller/front/ChapterCommentController.java
package com.novel.controller.front;

import cn.dev33.satoken.annotation.SaCheckLogin;
import com.novel.dto.req.SegmentCommentPageQueryReq;
import com.novel.po.chapterSummary.ChapterSummary;
import com.novel.po.comment.BookComment;
import com.novel.po.comment.SegmentComment;
import com.novel.result.PageResult;
import com.novel.result.Result;
import com.novel.service.ChapterCommentService;
import com.novel.user.dto.comment.AddChapterComment;
import com.novel.user.dto.comment.SegmentCommentView;
import com.novel.utils.IpUtils;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * 章节段落控制器
 * @author YYJY
 */
@Tag(name = "章节评论")
@RequestMapping("/chapter-comment")
@RestController
public class ChapterCommentController {

	@Autowired
	private ChapterCommentService service;
	
	@Operation(summary = "获取章节的概要列表信息")
	@GetMapping("summary")
	public Result<List<ChapterSummary>> getCommentSummaryList(long bookId, long chapterId) {
		List<ChapterSummary> list = service.getCommentSummaryList(bookId, chapterId);
		return Result.ok(list);
	}
	
	@SaCheckLogin
	@Operation(summary = "添加章节段落评论")
	@PostMapping("add")
	public Result<SegmentCommentView> addChapterSummary(@RequestBody AddChapterComment comment, HttpServletRequest request) {
		String ip = IpUtils.getIp(request);
		String address = IpUtils.getIpAddress(ip);
		return Result.ok(service.addChapterSummary(ip, address,comment));
	}
	
	@Operation(summary = "获取段落评论")
	@GetMapping("list")
	public Result<PageResult<SegmentCommentView>> listSegmentComment(SegmentCommentPageQueryReq query) {
		return Result.ok(service.listSegmentComment(query));
	}
	
	@Operation(summary = "点赞 - 未完成")
	@PostMapping("like")
	public Result<Void> like(long commentId) {
		service.like(commentId);
		return Result.ok();
	}
	
	@Operation(summary = "删除评论")
	@DeleteMapping("delete")
	public Result<Void> delete(long commentId) {
		service.delete(commentId);
		return Result.ok();
	}
	
}

File: frontend-nuxt/app/__generated/services/ChapterCommentController.ts
import type {Executor} from '../';
import type {Dynamic_ChapterSummary} from '../model/dynamic/';
import type {
    AddChapterComment, 
    PageResult, 
    Result, 
    SegmentCommentPageQueryReq, 
    SegmentCommentView, 
    Void
} from '../model/static/';

/**
 * 章节段落控制器
 */
export class ChapterCommentController {
    
    constructor(private executor: Executor) {}
    
    readonly addChapterSummary: (options: ChapterCommentControllerOptions['addChapterSummary']) => Promise<
        Result<SegmentCommentView>
    > = async(options) => {
        let _uri = '/chapter-comment/add';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<Result<SegmentCommentView>>;
    }
    
    readonly delete: (options: ChapterCommentControllerOptions['delete']) => Promise<
        Result<Void>
    > = async(options) => {
        let _uri = '/chapter-comment/delete';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.commentId;
        _uri += _separator
        _uri += 'commentId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'DELETE'})) as Promise<Result<Void>>;
    }
    
    readonly getCommentSummaryList: (options: ChapterCommentControllerOptions['getCommentSummaryList']) => Promise<
        Result<ReadonlyArray<Dynamic_ChapterSummary>>
    > = async(options) => {
        let _uri = '/chapter-comment/summary';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.bookId;
        _uri += _separator
        _uri += 'bookId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        _value = options.chapterId;
        _uri += _separator
        _uri += 'chapterId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<ReadonlyArray<Dynamic_ChapterSummary>>>;
    }
    
    readonly like: (options: ChapterCommentControllerOptions['like']) => Promise<
        Result<Void>
    > = async(options) => {
        let _uri = '/chapter-comment/like';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.commentId;
        _uri += _separator
        _uri += 'commentId='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'POST'})) as Promise<Result<Void>>;
    }
    
    readonly listSegmentComment: (options: ChapterCommentControllerOptions['listSegmentComment']) => Promise<
        Result<PageResult<SegmentCommentView>>
    > = async(options) => {
        let _uri = '/chapter-comment/list';
        let _separator = _uri.indexOf('?') === -1 ? '?' : '&';
        let _value: any = undefined;
        _value = options.query.chapterId;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'chapterId='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.query.segmentNum;
        if (_value !== undefined && _value !== null) {
            _uri += _separator
            _uri += 'segmentNum='
            _uri += encodeURIComponent(_value);
            _separator = '&';
        }
        _value = options.query.pageNum;
        _uri += _separator
        _uri += 'pageNum='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        _value = options.query.pageSize;
        _uri += _separator
        _uri += 'pageSize='
        _uri += encodeURIComponent(_value);
        _separator = '&';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<PageResult<SegmentCommentView>>>;
    }
}

export type ChapterCommentControllerOptions = {
    'getCommentSummaryList': {
        readonly bookId: number, 
        readonly chapterId: number
    }, 
    'addChapterSummary': {
        readonly body: AddChapterComment
    }, 
    'listSegmentComment': {
        readonly query: SegmentCommentPageQueryReq
    }, 
    'like': {
        readonly commentId: number
    }, 
    'delete': {
        readonly commentId: number
    }
}
